src/scene/graph-node.js

```diff
@@ -913,16 +917,32 @@ Object.assign(pc, function () {
 
         _dirtifyWorld: function () {
             if (!this._dirtyWorld) {
-                this._dirtyWorld = true;
-                for (var i = 0; i < this._children.length; i++) {
-                    if (!this._children[i]._dirtyWorld)
-                        this._children[i]._dirtifyWorld();
-                }
+                this._queueSync();
+                if (!this._dirtyWorld)
+                    this._dirtifyWorldImpl();
+            }
+        },
+
+        _dirtifyWorldImpl: function () {
+            this._dirtyWorld = true;
+            for (var i = 0; i < this._children.length; i++) {
+                if (!this._children[i]._dirtyWorld)
+                    this._children[i]._dirtifyWorldImpl();
             }
             this._dirtyNormal = true;
             this._aabbVer++;
         },
 
+        // Sync Queue is currently in App and there is no reference to App obj in graph-node
+        // It's temporary solution to access it from global namespace
+        _queueSync: function () {
+            pc.Application.getApplication().syncQueue.push(this._graphDepth, this);
```

Comment:
I'm not super-keen about grabbing the application like this. What do you think, @daredevildave? I do also kinda feel like the sync queue should be on the pc.Scene, not the application.